pkg/cloudprovider/cloudprovider.go

```diff
@@ -337,3 +348,16 @@ func (c *CloudProvider) instanceToMachine(ctx context.Context, ec2instance *ec2.
 	machine.Status.ProviderID = fmt.Sprintf("aws:///%s/%s", aws.StringValue(ec2instance.Placement.AvailabilityZone), aws.StringValue(ec2instance.InstanceId))
 	return machine
 }
+
+func (c *CloudProvider) getInstance(ctx context.Context, id string) (*ec2.Instance, error) {
+	instanceID, err := utils.ParseInstanceID(id)
+	if err != nil {
+		return nil, err
+	}
+	ctx = logging.WithLogger(ctx, logging.FromContext(ctx).With("id", id))
```

Comment:
This updated context won't be held outside of this function call. Should we make sure that we update the logging context outside of `getInstance` as well?